src/main.rs

```diff
@@ -141,7 +141,7 @@ fn main() {
     let this_pkg = res.packages.last().unwrap();
     let metadata_value = this_pkg
         .metadata
-        .get("nanos")
+        .get(device_str)
```

Comment:
There's only a single expected metadata section `[packages.metadata.nanos]` for now. I know it's not appropriately named though, since it applies to all devices (s/x/s+,...)
Defining device-specific options in the future will be possible with a variant of what you propose, but as it is it wont work for builds other than `nanos`

Reply:
I need to support at least Nano S and Nano S+ builds. With the change above, I can generate manifests specific to device type (with different icons and other stuff). What else I'm missing?